applications/inkjet.md

```diff
@@ -0,0 +1,280 @@
+
+Analysis will have a button and results. The button will take the current smart contract (must be open in VS Code current tab), and run it through CoinFabrik's Scout. If Scout is not installed, it will notify the user, and install it if they wish. After the vulnerability analysis is ran, the results will be displayed in the results part of this section.
+
+Templates will have a simple dropdown, where the user can select a template to use. There will be a "Create" button, where when clicked, a new `.rs` file will be initiated in the current file directory with the template contract within it.
+
+
+#### RAG - LLM Architecture
+
+![architecture](https://jyu.llc/inkjet_arch.png)
+
+#### Stack
+
+We will be using the following technologies:
+- Typescript/JS and Node.js (Electron Framework) for VS Code Extension
+- Javascript for RAG-LLM pipeline
+ - LlamaIndex and LangChain libraries for data loading, processing, embedding
+ - LlamaIndex and LangChain libraries for vectorstore retrieval and LLM interaction with retrieval results
+- Milvus and Weaviate for Vectorstore
+- OpenAI text-embedding-ada-002 Model for Embeddings
+- OpenAI for LLM (GPT-4-32k, GPT-4-1106-preview)
+- ink!/Rust for Smart Contracts
+- CoinFabrik Scout for Vulnerabiity Detection
+
+#### ink! Smart Contract Vectorstore:
+
+For our vectorstore dataset, we will source from a variety of areas. We will scrape the source code from verified contracts on ink!/WASM enabled chains, such as Astar and Aleph Zero. Github repos of example contracts will be scraped, such as [paritytech](https://github.com/paritytech/ink-examples/tree/main), [Astar WASM Showcase](https://github.com/AstarNetwork/wasm-showcase-dapps/tree/main), [Metis](https://github.com/patractlabs/metis), and others.
+
+To compliment these examples, we will also scrape all ink! documentations available from the [ink! docs](https://use.ink/), [Substrate docs](https://docs.substrate.io/tutorials/smart-contracts/develop-a-smart-contract/), and others.
+
+Additionally, we will use a barebones RAG-LLM pipeline with only documentation loaded to create generated examples through guided prompts. We divide our approach into categories and subcategories of smart contract purposes, including payments, transfers, lending, borrowing, vesting, escrow, NFTs, tokens, supply chain management, invoicing, real estate ownership, DAOs, decentralized identity, gaming mechanics, auctions, reputation systems, etc.
+
+For each subcategory, we will generate at least 100 distinct examples, which should be sufficient for a diverse representation of the general purposes of smart contracts. Before adding each smart contract to the dataset, we will run, compile, test with CoinFabrik, and deploy to a local or testnet node to ensure functionality and security.
```

Comment:
Can you describe in detail how you're going to ensure security? Is this a manual review or are you using specific techniques?

Reply:
We are planning to use CoinFabrik Scout to detect vulnerabilities, and manually debug if they exist until it passes the audit.